pkg/login/social/generic_oauth.go

```diff
@@ -412,33 +434,75 @@ func (s *SocialGenericOAuth) FetchPrivateEmail(client *http.Client) (string, err
 	return email, nil
 }
 
-func (s *SocialGenericOAuth) FetchTeamMemberships(client *http.Client) ([]int, bool) {
+// FetchTeamMemberships will attempt to get the user's team string IDs from the JWT claims using the 'team_id_attribute_path' setting.
+// If the attribute path does not find anything in the JWT claims, then the same attribute path is used to attempt to retrieve the team IDs from the OIDC '/userinfo' endpoint.
+// If the attribute path still does not find anything in the '/userinfo' response, then the same attribute path is used to attempt to retrieve the team IDs from the 'teams_url'
+// If team IDs are found, they are returned. Otherwise, an error is returned.
+func (s *SocialGenericOAuth) FetchTeamMemberships(client *http.Client) ([]string, error) {
+	var err error
+	var ids []string
+
+	if s.teamsUrl == "" {
+		ids, err = s.fetchTeamMembershipsFromDeprecatedTeamsUrl(client)
```

Comment:
It adds some complexity having two branches for fetching team memberships. Have you considered just fallback the `teamsUrl` to the behavior before these changes and setting the `teamIdAttributePath` to a default value and then only use the `fetchTeamMembershipsFromTeamsUrl`? 

I'm not sure it would work, but I tested the following on jmespath.org:
```json
[
 { "id": 1 },
 { "id": 2 }
]
```
The expression `[*].to_string(id)` produces the following result which seems like it can be used with `fetchTeamMembershipsFromTeamsUrl`:
```json
[
  "1",
  "2"
]
```

This would allow us to reuse the same code for extracting team ids and it would be completely backward compatible for everyone. WDYT?

Reply:
@marefr discussed this and we are comfortable with the current logic and making it clear that one path is the "old" path.